src/south/SQLDbToFile/SQLDbToFile.class.js

```diff
@@ -429,6 +476,58 @@ class SQLDbToFile extends ProtocolHandler {
     }
     return csv.unparse(result, options)
   }
+
+  /**
+   * Log the executed query with replacements
+   * @param {string} query - The query
+   * @param {Date} startTime - The replaced StartTime
+   * @param {Date} endTime - The replaced EndTime
+   * @param {number} maxReturnValues - The replaced MaxReturnValues
+   * @returns {void}
+   */
+  logQuery(query, startTime, endTime, maxReturnValues) {
+    const startTimeLog = query.indexOf('@StartTime') !== -1 ? `StartTime = ${startTime.toISOString()}` : ''
+    const endTimeLog = query.indexOf('@EndTime') !== -1 ? `EndTime = ${endTime.toISOString()}` : ''
+    const maxReturnValuesLog = query.indexOf('@MaxReturnValues') !== -1 ? `MaxReturnValues = ${maxReturnValues}` : ''
+    this.logger.debug(`Executing "${query}" with ${startTimeLog} ${endTimeLog} ${maxReturnValuesLog}`)
+  }
+
+  /**
+   * Generate replacements parameters
+   * @param {string} query - The query
+   * @param {Date} startTime - The StartTime
+   * @param {Date} endTime - The EndTime
+   * @param {number} maxReturnValues - The MaxReturnValues
+   * @return {any[]} - The replacement parameters
+   */
+  static generateReplacementParameters(query, startTime, endTime, maxReturnValues) {
+    const startTimeOccurrences = SQLDbToFile.getOccurrences(query, '@StartTime', startTime)
+    const endTimeOccurrences = SQLDbToFile.getOccurrences(query, '@EndTime', endTime)
+    const maxReturnValuesOccurrences = SQLDbToFile.getOccurrences(query, '@MaxReturnValues', maxReturnValues)
+    const occurrences = startTimeOccurrences.concat(endTimeOccurrences).concat(maxReturnValuesOccurrences)
+    occurrences.sort((a, b) => (a.index - b.index))
+    return occurrences.map((occurrence) => occurrence.value)
+  }
+
+  /**
+   * Get all occurrences of a substring with a value
+   * @param {string} str - The string to look for occurrences in
+   * @param {string} keyword - The keyword
+   * @param {any} value - The value to assign to the occurrences index
+   * @return {object[]} - The result as { index, value}
+   */
+  static getOccurrences(str, keyword, value) {
```

Comment:
Can you add specific tests and examples for this method like you did for formatDateWithTimezone method ?